lib/measures/tbd/measure.rb

```diff
@@ -296,6 +296,41 @@ def run(mdl, runner, args)
       end
     end
 
+    # Pre-validate ground-facing constructions for KIVA.
+    if argh[:kiva_force] || argh[:gen_kiva]
+      kva = true
+
+      mdl.getSurfaces.each do |s|
+        id = s.nameString
+        construction = s.construction
+        next unless s.isGroundSurface
+
+        if construction.empty?
+          runner.registerError("Invalid construction for KIVA (#{id})")
+          kva = false if kva
+        else
+          construction = construction.get.to_LayeredConstruction
+
+          if construction.empty?
+            runner.registerError("KIVA requires layered constructions (#{id})")
+            kva = false if kva
+          else
+            construction = construction.get
+
+            unless TBD.standardOpaqueLayers?(construction)
+              runner.registerError("KIVA requires standard materials (#{id})")
+              kva = false if kva
+            end
+          end
+        end
+      end
+
+      unless kva
```

Comment:
If massless (or otherwise invalid) materials/constructions are targeted for KIVA input generation, raise informative ERROR messages and ignore user-required KIVA processes altogether.